lib/ex_aws/rds/impl.ex

```diff
@@ -0,0 +1,128 @@
+    query_params = Map.new
+    |> Map.put_new("Action", "ApplyPendingMaintenanceAction")
+    |> Map.put_new("ResourceIdentifier", resource_id)
+    |> Map.put_new("ApplyAction", action)
+    |> Map.put_new("OptInType", opt_in_type)
+    |> Map.put_new("Version", @version)
+
+    request(client, :post, "/", params: query_params)
+  end
+
+  def apply_pending_maintenance(client, resource_id, action, opt_in_type) when action == "db-upgrade" do 
+    query_params = Map.new
+    |> Map.put_new("Action", "ApplyPendingMaintenanceAction")
+    |> Map.put_new("ResourceIdentifier", resource_id)
+    |> Map.put_new("ApplyAction", action)
+    |> Map.put_new("OptInType", opt_in_type)
+    |> Map.put_new("Version", @version)
+
+    request(client, :post, "/", params: query_params)
+  end  
+
+  def describe_db_instances(client, opts \\ %{}) do 
+    query_params = Map.new
+    |> Map.put_new("Action", "DescribeDBInstances")
+    |> Map.put_new("Version", @version)
+    |> Map.merge(opts)
+
+    request(client, :get, "/", params: query_params)
+  end
+
+  def create_db_instance(client, instance_id, username, password, storage, class, engine, opts \\ %{}) do 
+    query_params = Map.new
+    |> Map.put_new("Action", "CreateDBInstance")
+    |> Map.put_new("MasterUsername", username)
+    |> Map.put_new("MasterUserPassword", password)
+    |> Map.put_new("AllocatedStorage", storage)
+    |> Map.put_new("DBInstanceIdentifier", instance_id)
+    |> Map.put_new("DBInstanceClass", class)
+    |> Map.put_new("Engine", engine)
+    |> Map.put_new("Version", @version)
+    |> Map.merge(opts)
+
+    request(client, :post, "/", params: query_params)
+  end
+
+  def modify_db_instance(client, instance_id, opts \\ %{}) do 
+    query_params = Map.new
+    |> Map.put_new("Action", "ModifyDBInstance")
+    |> Map.put_new("DBInstanceIdentifier", instance_id)
+    |> Map.put_new("Version", @version)    
+    |> Map.merge(opts)
+
+    request(client, :patch, "/", params: query_params)
+  end
+
+  def delete_db_instance(client, instance_id, final_snapshot_id \\ :empty, opts \\ %{}) do 
+    query_params = Map.new
+    |> Map.put_new("Action", "DeleteDBInstance")
+    |> Map.put_new("DBInstanceIdentifier", instance_id)
+    |> Map.put_new("Version", @version)
+
+    if final_snapshot_id != :empty do 
+      query_params = Map.put_new(query_params, "FinalDBSnapshotIdentifier", final_snapshot_id)
+    else
+      query_params = Map.put_new(query_params, "SkipFinalSnapshot", true)
+    end    
+
+    query_params = Map.merge(query_params, opts)
+
+    request(client, :delete, "/", params: query_params)
+  end
+
+  def reboot_db_instance(client, instance_id, failover \\ :empty) do
+    query_params = Map.new
+    |> Map.put_new("Action", "RebootDBInstance")
+    |> Map.put_new("DBInstanceIdentifier", instance_id)
+    |> Map.put_new("Version", @version)    
+
+    if failover != :empty do 
+      query_params = Map.merge(query_params, %{"Failover" => failover})
```

Comment:
This needs refactored to avoid the imperative warning. IE

```elixir
query_params = if failover != :empty do
  Map.put(query_params, "Failover", failover)
else
  query_params
end
```